db/migrate/20160322195653_move_switch_host_to_jointable.rb

```diff
@@ -0,0 +1,25 @@
+class MoveSwitchHostToJointable < ActiveRecord::Migration[5.0]
+  class Host < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled # disable STI
+    has_many :switches,
+             :class_name => 'MoveSwitchHostToJointable::Switch'
+  end
+
+  class Switch < ActiveRecord::Base
+    belongs_to :host,
+               :class_name => 'MoveSwitchHostToJointable::Host'
+  end
+
+  class HostSwitch < ActiveRecord::Base
+  end
+
+  def up
+    say_with_time('Populating host_switches table with (host.id, switch.id)') do
+      Host.all.each do |host|
```

Comment:
So, it's good to use these techniques even when it's not required so you do it out of habit :wink: